File: output/CodeGen/avatar/Python/Java_repair/codeforces_459_A.java
import java.util.Scanner;
import java.lang.Math;

public class codeforces_459_A {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String[] str = in.nextLine().split(" ");
        int x1 = Integer.parseInt(str[0]);
        int y1 = Integer.parseInt(str[1]);
        int x2 = Integer.parseInt(str[2]);
        int y2 = Integer.parseInt(str[3]);
        int denominator = x2 - x1;
        int numerator = y2 - y1;
        if (denominator!= 0) {
            int quotient = numerator / denominator;
            if (numerator == 0) {
                int d = (int) Math.sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
                int x4 = x1;
                int x3 = x2;
                int y3 = y2 + d;
                int y4 = y1 + d;
                System.out.println(x3 + " " + y3 + " " + x4 + " " + y4);
            } else if (denominator == 0) {
                int y4 = y2;
                int y3 = y1;
                int d = (int) Math.sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
                int x4 = x1 + d;
                int x3 = x2 + d;
                System.out.println(x3 + " " + y3 + " " + x4 + " " + y4);
            } else if (quotient == 1) {
                int x4 = x2;
                int x3 = x1;
                int y4 = y1;
                int y3 = y2;
                System.out.println(x3 + " " + y3 + " " + x4 + " " + y4);
            } else if (quotient == -1) {
                int x4 = x

File: output/ChatGPT/avatar/Python/C++/codeforces_459_A.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int x1, y1, x2, y2;
    cin >> x1 >> y1 >> x2 >> y2;

    int denominator = x2 - x1;
    int numerator = y2 - y1;
    double quotient;

    if (denominator != 0) {
        quotient = static_cast<double>(numerator) / denominator;
    }

    if (numerator == 0) {
        int d = sqrt(pow((x2 - x1), 2) + pow((y2 - y1), 2));
        int x4 = x1;
        int x3 = x2;
        int y3 = y2 + d;
        int y4 = y1 + d;
        cout << x3 << " " << y3 << " " << x4 << " " << y4 << endl;
    }
    else if (denominator == 0) {
        int y4 = y2;
        int y3 = y1;
        int d = sqrt(pow((x2 - x1), 2) + pow((y2 - y1), 2));
        int x4 = x1 + d;
        int x3 = x2 + d;
        cout << x3 << " " << y3 << " " << x4 << " " << y4 << endl;
    }
    else if (quotient == 1) {
        int x4 = x2;
        int x3 = x1;
        int y4 = y1;
        int y3 = y2;
        cout << x3 << " " << y3 << " " << x4 << " " << y4 << endl;
    }
    else if (quotient == -1) {
        int x4 = x1;
        int x3 = x2;
        int y4 = y2;
        int y3 = y1;
        cout << x3 << " " << y3 << " " << x4 << " " << y4 << endl;
    }
    else {
        cout << "-1" << endl;
    }

    return 0;
}